pkg/e2ecluster/e2ehelpers/kind.go

```diff
@@ -28,15 +29,17 @@ func MaybeCreateTempKindCluster(namePrefix string) env.Func {
 			if err != nil {
 				return ctx, err
 			}
+			// Automatically clean up the cluster when the test finishes
+			testenv.Finish(deleteTempKindCluster())
 			return context.WithValue(ctx, helperContextKey(keyTempClusterName), name), nil
```

Comment:
With this change we don't need to store the cluster name in the context any more. It can be a local variable here and passed in to `deleteTempKindCluster`.